### PR TITLE
Add SPI Full Duplex DMA test

### PR DESCRIPTION
**Pull request by SergioGasquez**
## Thank you for your contribution!

We appreciate the time and effort you've put into this pull request.
To help us review it efficiently, please ensure you've gone through the following checklist:

### Submission Checklist 📝
- [x] I have updated existing examples or added new ones (if applicable).
- [x] My changes were added to the [`CHANGELOG.md`](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/CHANGELOG.md) in the **_proper_** section.
#### Extra:
- [x] I have read the [CONTRIBUTING.md guide](https://github.com/esp-rs/esp-hal/blob/main/CONTRIBUTING.md) and followed its instructions.

#### Description
Add SPI Full Duplex DMA tests

As with AES DMA, I tried to use Context as with other tests, but Rust compiler just wasn't happy with the types and all the movements of them, so I ended up copy/pasting it all over, which makes it really ugly.

#### Testing
Tested locally on C3, C6, H2 and S3.

Manually triggered the HIL workflow: https://github.com/esp-rs/esp-hal/actions/runs/8701948778

### CODE REVIEW

File: hil-test/tests/spi_full_duplex_dma.rs [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+        send.copy_from_slice(&[0xde, 0xad, 0xbe, 0xef]);
+
+        let transfer = spi.dma_transfer(&mut send, &mut receive).unwrap();
+        transfer.wait().unwrap();
+        assert_eq!(send, receive);
+    }
+
+    #[test]
+    #[timeout(3)]
+    fn test_asymestric_dma_transfer() {
```
Comment (bjoernQ) [inline hil-test/tests/spi_full_duplex_dma.rs:85]: nit-pick: typo -> `test_asymetric_dma_transfer`

**Review by bjoernQ — APPROVED** [review]
LGTM!

File: hil-test/tests/spi_full_duplex_dma.rs (hunk 1 above)
Comment (SergioGasquez) [reply hil-test/tests/spi_full_duplex_dma.rs]: Good catch! the typo was on all the test haha

**Review by MabezDev — APPROVED** [review]
Thanks!

**Pull request merged**